frontend/src/metabase/nav/containers/AppBar/AppBar.unit.spec.tsx

```diff
@@ -0,0 +1,200 @@
+        expect(screen.getByTestId("sidebar-toggle")).toBeVisible();
+
+        screen.getByTestId("sidebar-toggle").click();
+        expect(screen.getByText(/Our analytics/)).not.toBeVisible();
+        expect(screen.queryByTestId("main-logo")).not.toBeInTheDocument();
+        expect(screen.getByTestId("sidebar-toggle")).toBeVisible();
+      });
+
+      it("should not show either logo or side nav toggle button at all", async () => {
+        renderAppBar({
+          side_nav: false,
+          logo: false,
+        });
+
+        expect(await screen.findByText(/Our analytics/)).toBeVisible();
+        expect(screen.queryByTestId("main-logo")).not.toBeInTheDocument();
+        expect(screen.queryByTestId("sidebar-toggle")).not.toBeInTheDocument();
+      });
+    });
+
+    describe("small screens", () => {
+      beforeEach(() => {
+        matchMediaSpy.mockReturnValue(getMediaQuery({ matches: true }));
+      });
+
+      it("should be able to toggle side nav", async () => {
+        renderAppBar({
+          side_nav: true,
+        });
+
+        expect(await screen.findByText(/Our analytics/)).toBeVisible();
+        expect(screen.getByTestId("main-logo")).toBeVisible();
+
+        screen.getByTestId("sidebar-toggle").click();
+        expect(screen.queryByText(/Our analytics/)).not.toBeInTheDocument();
+        expect(screen.getByTestId("main-logo")).toBeVisible();
+      });
+
+      it("should hide side nav toggle icon", async () => {
+        renderAppBar({
+          side_nav: false,
+        });
+
+        expect(await screen.findByText(/Our analytics/)).toBeVisible();
+        expect(screen.getByTestId("main-logo")).toBeVisible();
+        expect(screen.queryByTestId("sidebar-toggle")).not.toBeInTheDocument();
+      });
+
+      it("should always show side nav toggle icon when logo is hidden", async () => {
+        renderAppBar({
+          side_nav: true,
+          logo: false,
+        });
+
+        expect(await screen.findByText(/Our analytics/)).toBeVisible();
+        expect(screen.queryByTestId("main-logo")).not.toBeInTheDocument();
+        expect(screen.getByTestId("sidebar-toggle")).toBeVisible();
+
+        screen.getByTestId("sidebar-toggle").click();
+        expect(screen.queryByText(/Our analytics/)).not.toBeInTheDocument();
+        expect(screen.queryByTestId("main-logo")).not.toBeInTheDocument();
+        expect(screen.getByTestId("sidebar-toggle")).toBeVisible();
+      });
+
+      it("should not show either logo or side nav toggle button at all", async () => {
+        renderAppBar({
+          side_nav: false,
+          logo: false,
+        });
+
+        expect(await screen.findByText(/Our analytics/)).toBeVisible();
+        expect(screen.queryByTestId("main-logo")).not.toBeInTheDocument();
+        expect(screen.queryByTestId("sidebar-toggle")).not.toBeInTheDocument();
+      });
+    });
+  });
+});
+
+function renderAppBar(embedOptions: Partial<EmbedOptions>) {
```

Comment:
Could you use the standard `setup()` pattern for testing? Like here https://github.com/metabase/metabase/blob/1bcbe954e3f7fda3703e4f6080ec86f7dc60e55c/frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceModal.unit.spec.tsx#L184. cc @kulyk

Reply:
That's fair, I just don't like that the name doesn't convey anything specific, and it doesn't look like it follows the normal pattern (arrange, act, assert/given when then) as `setup` sounds more like it should belong to the arrange/given section of the test.

I do realize, however, that we use `setup` in a number of places. I'll follow our defacto convention now, but this is worth discussing in the guild meeting maybe.